From 90a7bebedad1bc247c9f9dfd692a2e20aaf5855b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 28 Jan 2021 15:34:18 +0200
Subject: [PATCH] [SW-28801] gaudi: reset: align to the latest

Align the global reset sequence to the latest and greatest.
This involves changing the places between the halt engines and the PCIe
RRs closure.

Change-Id: Id48a21ee8a11e4bf13c539f71b49c95a8d003009
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/reset.c b/gaudi/reset.c
index 86d680c..334a38d 100644
--- a/gaudi/reset.c
+++ b/gaudi/reset.c
@@ -382,6 +382,7 @@
 
 	udelay(1000);
 
+	/* The reset mask is set by BTL, so we do not need to do this here */
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_SW_ALL_RST);
 }
 
@@ -463,8 +464,8 @@
 
 void hl_dev_reset(void)
 {
-	gaudi_close_pcie_rr();
 	gaudi_halt_engines();
+	gaudi_close_pcie_rr();
 	gaudi_pcie_disable_access();
 	gaudi_reset_soc_initiators();
 
